code/modules/antagonists/changeling/changeling.dm

```diff
@@ -519,6 +524,31 @@
 	hud.leave_hud(owner.current)
 	set_antag_hud(owner.current, null)
 
+/**
+ * Signal proc for [COMSIG_MOB_MIDDLECLICKON] and [COMSIG_MOB_ALTCLICKON].
+ * Allows the changeling to sting people with a click.
+ */
+/datum/antagonist/changeling/proc/on_click_sting(mob/living/ling, atom/clicked)
+	SIGNAL_HANDLER
+
+	// nothing to handle
+	if(!chosen_sting)
+		return
+
+	if(!isliving(ling) || clicked == ling || ling.stat != CONSCIOUS)
+		return
+
+	// sort-of hack done here: we use in_given_range here because it's quicker.
+	// actual ling stings do pathfinding to determine whether the target's "in range".
+	// however, this is "close enough" preliminary checks to not block click
+	if(!isliving(clicked) || !IN_GIVEN_RANGE(ling, clicked, sting_range))
+		return
```

Comment:
```suggestion

 // Checks if the target is within the sting range of the changeling.
/proc/in_changeling_sting_range(mob/living/ling, mob/living/target, range)
	return (get_dist(ling, target) <= range)

// Using a more specific proc for the changeling sting range check.
// Actual ling stings do pathfinding to determine whether the target is "in range".
// This is a "close enough" preliminary check to not block click.
if(!isliving(clicked) || !in_changeling_sting_range(ling, clicked, sting_range))
	return
```

Might as well just do it in a non hack way

Reply:
This is literally what it's already doing but as a proc instead